tests/common-test-functionality.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
```

Comment:
I wonder whether it would be more simple to write this script in python for later maintenance instead of a bash script and how much effort it would take?
If you think this is feasible we can do that in a follow-up PR

Reply:
OK, sounds good, let's make a python script instead of this bash script in a follow-up PR